src/ORM/Table.php

```diff
@@ -1011,6 +1011,14 @@ public function findAll(Query $query, array $options)
      */
     public function findList(Query $query, array $options)
     {
+        if (!isset($options['keyField']) && !isset($options['valueField'])) {
+            $select = $query->clause('select');
+            if ($select && count($select) <= 2) {
+                $options['keyField'] = array_shift($select);
+                $options['valueField'] = array_shift($select) ?: $options['keyField'];
```

Comment:
OK, in those cases one has to add the key and value field definitions.
But run my tests without my patch and you will see that it uses two totally different fields and resulting in null rows without the expectation of it.
If you later at least would run into some error it would be clear what is going on.
But currently fields are silently changed without the intention to do so.